### PR TITLE
[Feature](auto-inc) Add start value for auto increment column

### PR DESCRIPTION
**Pull request by Yukang-Lian**
## Proposed changes

Issue Number: close #xxx

```mysql
CREATE TABLE IF NOT EXISTS `test` (
              `id` BIGINT NOT NULL AUTO_INCREMENT(100),
              `value` int NOT NULL
            ) ENGINE=OLAP
            DUPLICATE KEY(`id`)
            DISTRIBUTED BY HASH(`id`) BUCKETS 1
            PROPERTIES (
            "replication_allocation" = "tag.location.default: 1"
            );

insert into test values(null,1);

select * from test;
+------+-------+
| id   | value |
+------+-------+
|  100 |     1 |
+------+-------+

insert into test (value) values(2);

select * from test;
+------+-------+
| id   | value |
+------+-------+
|  100 |     1 |
|  101 |     2 |
+------+-------+

```

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 1]
```diff
@@ -178,77 +178,79 @@ public String getValue() {
...
     public ColumnDef(String name, TypeDef typeDef, boolean isKey, AggregateType aggregateType,
-            boolean isAllowNull, boolean isAutoInc, DefaultValue defaultValue, String comment, boolean visible) {
+            boolean isAllowNull, long autoIncInitValue, DefaultValue defaultValue, String comment, boolean visible) {
         this.name = name;
         this.typeDef = typeDef;
         this.isKey = isKey;
         this.aggregateType = aggregateType;
         this.isAllowNull = isAllowNull;
-        this.isAutoInc = isAutoInc;
+        this.isAutoInc = autoIncInitValue != 0;
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:212]: should be  > 0 ?

File: fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java [hunk 2]
```diff
@@ -560,6 +562,11 @@ public String toSql() {
 
         if (isAutoInc) {
             sb.append("AUTO_INCREMENT ");
+            if (autoIncInitValue != 0) {
```
Comment (zhannngchen) [inline fe/fe-core/src/main/java/org/apache/doris/analysis/ColumnDef.java:565]: ditto

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run external

**Comment by Yukang-Lian** [comment]
run external

**Comment by Yukang-Lian** [comment]
run buildall

**Comment by Yukang-Lian** [comment]
run external

**Comment by Yukang-Lian** [comment]
run external

**Comment by Yukang-Lian** [comment]
run buildall

File: fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java [hunk 3]
```diff
@@ -2495,8 +2495,22 @@ public ColumnDefinition visitColumnDef(ColumnDefContext ctx) {
...
+                autoIncInitValue = Long.valueOf(ctx.autoIncInitValue.getText());
+                if (autoIncInitValue < 0) {
+                    throw new AnalysisException("AUTO_INCREMENT start value can not be negative.");
+                }
+            } else {
+                // AUTO_INCREMENT default 1.
+                autoIncInitValue = Long.valueOf(1);
+            }
+        } else {
+            autoIncInitValue = Long.valueOf(-1);
```
Comment (mymeiyi) [inline fe/fe-core/src/main/java/org/apache/doris/nereids/parser/LogicalPlanBuilder.java:2511]: The else can be removed

**Review by mymeiyi — APPROVED** [review]

**Comment by Yukang-Lian** [comment]
run buildall

**Review by zhannngchen — APPROVED** [review]
LGTM

**Pull request merged**